src/mongoose_graphql/admin/mongoose_graphql_stanza_admin_query.erl

```diff
@@ -0,0 +1,43 @@
+-module(mongoose_graphql_stanza_admin_query).
+
+-export([execute/4]).
+
+-ignore_xref([execute/4]).
+
+-include("../mongoose_graphql_types.hrl").
+-include("mongoose_logger.hrl").
+
+execute(_Ctx, _Obj, <<"getLastMessages">>, Opts) ->
+    get_last_messages(Opts).
+
+get_last_messages(#{<<"caller">> := Caller, <<"limit">> := Limit,
+                    <<"with">> := With, <<"before">> := Before})
+        when is_integer(Limit) ->
+    case mongoose_graphql_helper:check_user(Caller) of
+        {ok, _HostType} ->
+            get_last_messages2(Caller, Limit, With, Before);
+        Error ->
+            Error
+    end.
+
+get_last_messages2(Caller, Limit, With, Before) ->
+    With2 = null_as_undefined(With),
+    BeforeSeconds = maybe_datetime_to_seconds(Before),
+    Limit2 = min(500, Limit),
+    Rows = mongoose_stanza_api:lookup_recent_messages(Caller, With2, BeforeSeconds, Limit2),
+    Maps = lists:map(fun row_to_map/1, Rows),
+    {ok, #{<<"stanzas">> => Maps, <<"limit">> => Limit2}}.
+
+maybe_datetime_to_seconds(null) -> undefined;
+maybe_datetime_to_seconds(Microseconds) -> Microseconds / 1000000. %% Could be float
```

Comment:
Why do we convert it to float instead of just passing `Microseconds`? We divide by 1M just to multiply by 1M later.

Reply:
to be compatible with mod_commands way of calling actually (it passes before as seconds)